File: src/test/java/main/tenderInformation/TenderSettingsUpdate.java
package main.tenderInformation;

import main.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.Test;

import static org.testng.Assert.assertTrue;

public class TenderSettingsUpdate extends ApplicationsNavigationTest {

    @Test
    public void UpdateSubcontractorsSettings() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();

        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getTenderInformationTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToDirectoryFrame();
        TenderDescription tenderDescription = new TenderDescription(driver);
        tenderDescription.clickOnTenderSettingsSubTab();
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        tenderDescription.clickOnEditTenderSettingsButton();
        Thread.sleep(3000);
        tenderDescription.switchToNewWindowTest();
        TenderSettings tenderSettings = new TenderSettings(driver);
        tenderSettings.clickOnAccessDocumentsBySubcontractorDropdown();
        tenderSettings.clickOnlyTenderersMayDownloadDropdownOption();
       tenderSettings.clickOnAllowTendersPartnerToAccessDropdownOption();
        tenderSettings.subcontractorQAdropdown();
        tenderSettings.setNotifyAboutAnswersNo();
      tenderSettings.setNotifyAboutAnswersYes();
        tenderDescription.okButton().click();
    }
    @Test
    public void UpdateTenderSettingsOLD() throws InterruptedException {

        TendersPage tendersPage = new TendersPage(driver);

        WebElement iframeElement = driver.findElement(By.xpath("//frame[@name='browser']"));
        driver.switchTo().frame(iframeElement);
        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getTenderInformationTab());
        driver.switchTo().defaultContent(); // Повернення до головного вікна сторінки
// Знову "прив'язка" вебдрайвера до поточного потоку
        WebDriverWait driverWait = new WebDriverWait(driver, 3); // Зміна wait для очікування
        driverWait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//frame[@name='directory']")));
        WebElement frameDirectory = driver.findElement(By.xpath("//frame[@name='directory']"));
        driver.switchTo().frame(frameDirectory);
        TenderDescription tenderDescription = new TenderDescription(driver);
        tenderDescription.clickOnTenderSettingsSubTab(); // Продовження взаємодії з елементами в фреймі
        driver.switchTo().defaultContent(); // Повернення до головного вікна сторінки
        WebElement frameNavigation = driver.findElement(By.xpath("//frame[@name='navigation']"));
        driver.switchTo().frame(frameNavigation);
        tenderDescription.clickOnEditTenderSettingsButton();
        tenderDescription.switchToNewWindowTest();
        Thread.sleep(3000);
        TenderSettings tenderSettings = new TenderSettings(driver);
        tenderSettings.clickOnAccessDocumentsBySubcontractorDropdown();
        tenderSettings.clickOnlyTenderersMayDownloadDropdownOption();
        tenderDescription.okButton().click();
    }


}

File: src/test/java/main/tenderInformation/TenderSettingsUpdate.kt
package main.tenderInformation

import main.ApplicationsNavigationTest
import main.PublicTenderPage
import main.TendersPage
import main.users.all.Administrators
import org.testng.annotations.BeforeMethod
import org.testng.annotations.Test

class TenderSettingsUpdate : ApplicationsNavigationTest() {
    private lateinit var tenderDescription: TenderDescription

    @BeforeMethod
    fun setUpTest() {
    val tendersPage = TendersPage(driver)
        val tenderDescription = TenderDescription(driver!!)

        tendersPage.switchToBrowserFrame()

    val publicTenderPage = PublicTenderPage(driver)
    publicTenderPage.tenderInformationTab?.click()
    driver!!.switchTo().defaultContent()
    tendersPage.switchToDirectoryFrame()
    tenderDescription.clickOnTenderSettingsSubTab()
    driver!!.switchTo().defaultContent()
    tendersPage.switchToNavigationFrame()
    tenderDescription.clickOnEditTenderSettingsButton()
    Thread.sleep(3000)
    tenderDescription.switchToNewWindowTest()}

        @Test
    @Throws(InterruptedException::class)
    fun UpdateSubcontractorsSettings() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.clickOnAccessDocumentsBySubcontractorDropdown()
        tenderSettings.clickOnlyTenderersMayDownloadDropdownOption()
        tenderSettings.clickOnAllowTendersPartnerToAccessDropdownOption()
        tenderSettings.subcontractorQAdropdown()
        tenderSettings.setNotifyAboutAnswersNo()
        tenderSettings.setNotifyAboutAnswersYes()
        tenderDescription.okButton().click()
    }

    @Test
    @Throws(InterruptedException::class)
    fun selectSupplementaryFilesAcceptedTest() {
        val tenderSettings = TenderSettings(driver)
        tenderSettings.selectSupplementaryFilesAccepted()
        tenderDescription.okButton().click() }

    @Test
    @Throws(InterruptedException::class)
    fun selectSupplementaryFilesNotAcceptedTest() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.selectSupplementaryFilesNoAccepted()
        Thread.sleep(5000)
        tenderDescription.okButton().click() }

    @Test
    @Throws(InterruptedException::class)
    fun UpdateSecuritySettings() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.updateSecuritySettings()
        tenderDescription.okButton().click()}

    @Test
    @Throws(InterruptedException::class)
    fun UpdateOpeningAndPublicationSettings() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.clickOnAccessDocumentsBySubcontractorDropdown()
        tenderSettings.clickOnlyTenderersMayDownloadDropdownOption()
        tenderSettings.clickOnAllowTendersPartnerToAccessDropdownOption()
        tenderSettings.subcontractorQAdropdown()
        tenderSettings.setNotifyAboutAnswersNo()
        tenderSettings.setNotifyAboutAnswersYes()
        tenderDescription.okButton().click() }

    @Test
    @Throws(InterruptedException::class)
    fun UpdateQandAHadlingSetting() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.updateQAhandlingNoYes()
        tenderDescription.okButton().click() }

    @Test
    @Throws(InterruptedException::class)
    fun UpdateSendersVisibilitySettings() {
        val tenderSettings = TenderSettings(driver!!)
        tenderSettings.updateSendersVisibilitySettings()
        tenderDescription.okButton().click()
    }
}
